src/test/java/com/emeq/counter/controller/FoodsControllerTest.java

```diff
@@ -0,0 +1,164 @@
+package com.emeq.counter.controller;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import com.emeq.counter.dto.FoodDTO;
+import com.emeq.counter.exception.RecordNotFoundException;
+import com.emeq.counter.model.Food;
+import com.emeq.counter.repository.FoodRepository;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+class FoodsControllerTest {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private FoodsController controller;
+
+    @MockBean
+    private FoodRepository foodRepository;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Test
+    public void contexLoads() throws Exception {
+        assertThat(controller).isNotNull();
+    }
+
+    @Test
+    public void allFoods() {
+        Food foods[] = { createFood(1, "banana"), createFood(2, "apple"), createFood(3, "orange") };
+        Iterable<Food> iterable = Arrays.asList(foods);
+
+        when(foodRepository.findAll()).thenReturn(iterable);
+
+        ResponseEntity<FoodDTO[]> response = this.restTemplate.getForEntity(endpoint("/foods"), FoodDTO[].class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+
+        assertThat(mapper.convertValue(response.getBody(), JsonNode.class))
+                .isEqualTo(mapper.convertValue(convertDtos(iterable), JsonNode.class));
```

Comment:
why does it have 8-9 spaces?

Reply:
There are two tabs ( or 8 space). We decided with @brckks to use two tabs indentations when chain methods.